### PR TITLE
Added more specs

### PR DESCRIPTION
**Pull request by franchester4**
Closes: https://snapsheettech.atlassian.net/browse/SNAPTX-3828

Added a more detailed test suite in `specs/ssm_config_yml_spec.rb`. YAML files used to test ssm_config are in `spec/fixtures`.

Changed `ssm_config.rb` so that the config path is now a const, allows for easier testing + removes reliance on magic string

Added a basic rails app to specs in order for RSpec to work (required Rails in a non rails app, so adding was necessary)

Added `rails_helper.rb` to help with testing as well

Added RuboCop + necessary adjustments to Gemfile

### CODE REVIEW

File: .DS_Store
Comment (lorman) [inline .DS_Store:1]: Can you add this to the '.gitignore' file?

File: spec/rails_app/log/development.log
Comment (lorman) [inline spec/rails_app/log/development.log:1]: We can/should add the `log` directory to the `.gitignore`

File: spec/ssm_config_yml_spec.rb
Comment (lorman) [inline spec/ssm_config_yml_spec.rb:1]: Can you fix tabs and indentation here please? 2 spaces should be our default. We can also look into adding rubocop to the repo dev requirements to help keep it clean.

**Review by holmes2 — COMMENTED** [review]
All the files with symbol ⛔ at the end has no newline  at the end. Can you add that as well ?

File: .DS_Store
Comment (lorman) [reply .DS_Store:1]: Might need to manually remove this file and the logs from this commit. You can probaby just `git rm` to delete the files all together.

File: spec/ssm_config_yml_spec.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+  context "When a YAML file exists" do
+    context "when YAML file is blank" do
+      let(:no_method_error_message) { "undefined method `[]' for false:FalseClass" }
+      context "with 0 depth" do
+        it "raises NoMethodError"
+          expect{SsmConfig.blank}.to raise_error(NoMethodError).with_message(no_method_error_message)
+        end
+      end
+      context "with nontrivial depth" do
```
Comment (lorman) [inline spec/ssm_config_yml_spec.rb:16]: I think we typically have newlines between consecutive groups, context, it... please add a new line here and for some of the other examples.

**Review by lorman — APPROVED** [review]
CR: LG

**Comment by lorman** [comment]
@holmes2 should we make this a new minor version of the gem, or wait till we start making real changes to functionality?

**Review by lorman — APPROVED** [review]

**Comment by franchester4** [comment]
done

**Review by holmes2 — APPROVED** [review]
CR: ✅ 


Mr. Bean Thumbs Up

**Pull request merged**